com/bl/ttt/main.py

```diff
@@ -1,21 +1,88 @@
+                return new_game_input
+            else:
+                print('That was an invalid input: please try again.')
+                continue
+        except (Exception, ValueError):
+            break
+
+
+def get_game_type():
+    while True:
+        try:
+            game_type_input = input("What kind of game will you play: enter 'h' for human or 'c' for computer: ").upper()
+            if game_type_input is not None and (game_type_input == 'H' or game_type_input == 'C'):
+                return game_type_input
+            else:
+                print('That was an invalid input: please try again.')
+                continue
+        except (Exception, ValueError):
+            break
+
+
+def get_first_move(game):
     player_id, cell = input('Please enter player X or O, a space, and then a cell from 0 to 8: ').split(' ')
+    game.player1 = player_id
+    if game.player1 == 'X':
+        game.player2 = 'O'
+    else:
+        game.player2 = 'X'
     move = TictactoeMove(player_id, int(cell))
     if game.is_valid_move(move):
         game.make_valid_move(move)
     else:
-        print(f'Sorry, that is not a valid move for Player {player_id}, please try again.')
+        print(f'Sorry, that is not a valid move for Player {move.player_id}, please try again.')
         return player_tries_again
-    if game.has_won(player_id):
+    game.turn_player = player_id
+    game.turn_counter = 1
+    return move
+
+
+def process_another_move(game):
+    game.turn_counter += 1
+    if game.turn_player == 'X':
+        game.turn_player = 'O'
+    else:
+        game.turn_player = 'X'
+    player_id = game.turn_player
+    # print(f'GP1: {game.player1} | GP2: {game.player2} | GTP: {game.turn_player} | GCP: {game.computer_player} | GTC: {game.turn_counter}.')
+    if game.computer_player is not None and game.turn_counter % 2 == 0:
+        while True:
+            computer_cell = randrange(9)
+            move = TictactoeMove(player_id, computer_cell)
+            if game.is_valid_move(move):
+                game.make_valid_move(move)
+                print(f'The computer player {game.player2} is going to play cell {computer_cell}.')
+                time.sleep(2)
+            else:
+                continue
+            break
+    else:
+        while True:
+            human_cell = input(f'Player {player_id}: please enter a cell from 0 to 8: ')
+            move = TictactoeMove(player_id, int(human_cell))
+            if game.is_valid_move(move):
+                game.make_valid_move(move)
+            else:
+                print(f'Sorry, that is not a valid move for Player {move.player_id}, please try again.')
+                continue
+            break
+    if game.has_won(move.player_id):
         print(game)
-        print(f'Player {player_id} has won.')
+        print(f'Player {move.player_id} has won.')
         return game_over
     elif game.is_draw():
         print(game)
```

Comment:
Is there any way to break this function down into smaller parts to make it easier to read?
Somewhere there is a metric about 5 lines of code being optimal for a function - I follow that rule losely to be honest.